src/index.js

```diff
@@ -252,56 +259,65 @@ function UseWalletProvider({
         })
 
       if (!web3ReactConnector) {
-        throw new UnsupportedConnectorError(connectorId)
+        setStatus('error')
+        setError(new UnsupportedConnectorError(connectorId))
+        return
       }
 
       try {
         // TODO: there is no way to prevent an activation to complete, but we
         // could reconnect to the last provider the user tried to connect to.
-        setActivating(connectorId)
+        setConnector(connectorId)
         await web3ReactContext.activate(web3ReactConnector, null, true)
-        setActivated(connectorId)
-        setActivating(null)
+        setStatus('connected')
       } catch (err) {
-        setActivating(null)
-
         // Don’t throw if another connection has happened in the meantime.
         if (id !== activationId.current) {
           return
         }
+
+        // If not, the error has been thrown during the current connection attempt,
+        // so it's correct to indicate that there has been an error
+        setConnector(null)
+        setStatus('error')
+
         if (err instanceof UnsupportedChainIdError) {
-          throw new UnsupportedChainError(-1, chainId)
+          setError(new UnsupportedChainError(-1, chainId))
+          return
         }
         // It might have thrown with an error known by the connector
         if (connector.handleActivationError) {
-          connector.handleActivationError(err)
+          const thrownError = connector.handleActivationError(err)
+          setError(thrownError)
+          return
         }
```

Comment:
We should test if the error is actually handled by the connector here. Otherwise, having a connector with a `handleActivationError` method that doesn’t handle the error would return, and call `setError()` with `undefined` instead of `err`.

We could do this instead:

```jsx
const handledError = connector.handleActivationError?.(err)
if (handledError) {
  setError(handledError)
  return
}
```

Also changed `thrownError` to `handledError`, as it doesn’t get thrown anymore.

Reply:
Aah, missed this!